scripts/speech_recognition/code_switching/code_switching_audio_data_creation.py

```diff
@@ -0,0 +1,204 @@
+parser.add_argument("--manifest_path", default=None, type=str, help='Path to CS indermediate manifest')
+parser.add_argument(
+    "--audio_save_folder_path",
+    default=None,
+    type=str,
+    help='Path to directory where created synthetic set would be saved',
+)
+parser.add_argument("--manifest_save_path", default=None, type=str, help='Path to save the created manifest')
+parser.add_argument(
+    "--audio_normalized_amplitude", default=15000, type=int, help='Normalized amplitdue of audio samples'
+)
+parser.add_argument(
+    "--sample_beginning_pause_msec",
+    default=20,
+    type=int,
+    help='Pause to be added at the beginning of the sample (msec)',
+)
+parser.add_argument(
+    "--sample_joining_pause_msec",
+    default=100,
+    type=int,
+    help='Pause to be added between different phrases of the sample (msec)',
+)
+parser.add_argument(
+    "--sample_end_pause_msec", default=20, type=int, help='Pause to be added at the end of the sample (msec)'
+)
+
+args = parser.parse_args()
+
+
+def __read_manifest(manifest_path: str):
+    """
+    Args:
+        manifest_path: absolute path of the manifest file
+
+    Returns:
+        List with manifest entires as elements
+
+    """
+    data = []
+
+    for line in open(manifest_path, 'r'):
+        data.append(json.loads(line))
+
+    return data
+
+
+def __create_cs_data(
+    intermediate_cs_manifest_list: list,
+    audio_save_folder: str,
+    manfest_save_path: str,
+    audio_amplitude_normalization: int,
+    pause_beg_msec: int,
+    pause_join_msec: int,
+    pause_end_msec: int,
+):
+
+    """
+    Args:
+        intermediate_cs_manifest_list: the intermediate cs manifest obtained from code_switching_manifest_creation.py as a list
+        audio_save_folder: Absolute path to save the generated audio samples
+        manfest_save_path: Absolute path to save the corresponding manifest
+        audio_amplitude_normalization: The amplitude to scale to after normalization
+        pause_beg_msec: Pause to be added at the beginning of the sample (msec)
+        pause_join_msec: Pause to be added between different phrases of the sample (msec)
+        pause_end_msec: Pause to be added at the end of the sample (msec)
+
+    Returns:
+
+    """
+
+    sample_id = 0
+    fs = 16000
+    incorrect_sample_flag = 0
+
+    with open(manfest_save_path, 'w') as outfile:
+        for data in tqdm(intermediate_cs_manifest_list):
+
+            combined_audio = []
+
+            staring_pause = np.zeros(int(pause_beg_msec * fs / 1000))
+            combined_audio += list(staring_pause)
+
+            for index in range(len(data['lang_ids'])):
+
+                data_sample, fs_sample = librosa.load(data['paths'][index], sr=fs)
+                # Alternative-  fs_sample, data_sample = wavfile.read(data['paths'][index])
+
+                if fs_sample != 16000:
+                    print('!!ERROR!!!')
```

Comment:
better error handling? likely need to exit here? Also, use python / nemo logging!

Reply:
Why do we want to restrict to 16kHz? I think you should have an argument cs_data_sampling_rate which defaults to 16000. Input audio that is of a different sampling rate should be resampled to cs_data_sampling_rate.

Reply:
Actually the error check is not explicitly required here, it was incorrectly propagated from the development phase. [Librosa](https://librosa.org/doc/main/generated/librosa.load.html ) has been specially used because it allows us to specify the target sampling rate (sr=16000 in this case). This ensures that the input audio is always resampled to 16kHz, allowing us to handle datasets with varying sampling rates.
However, I would add an explicit check here for safety, exiting if the sampling rate condition is not met and adding the cause of the exit to the log.

Reply:
fixed